File: 1945-sum-of-digits-of-string-after-convert/1945-sum-of-digits-of-string-after-convert.java
class Solution {
    public int getLucky(String s, int k) {
        Map<Character,Integer> mp = new HashMap<>();
        int ct = 1 , res = 0;
        for(char c = 'a' ; c <= 'z' ; c++){
            mp.put(c ,ct);
            ct++;
        }
        StringBuilder inter = new StringBuilder();
        for(char ch : s.toCharArray()){
            inter.append(mp.get(ch));
        }
        String str = inter.toString();
        
        while(k > 0){
            res = 0;
            for(char ch : str.toCharArray()){
                res += (ch - '0');
            }
            str = String.valueOf(res);
            k--;
        }
        return res;
    }
}

File: 1945-sum-of-digits-of-string-after-convert/1945-sum-of-digits-of-string-after-convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        map<char,int> mp;
        int c = 1;
        for(char ch = 'a'; ch <= 'z' ; ch++){
            mp[ch] = c;
            c++;
        }
        string inter;
        for(char ch : s){
            inter += to_string(mp[ch]);
        }
        int res = 0;
        while(k--){
            res = 0;
            for(char ch : inter){
                res += (ch - '0');
            }
            inter = to_string(res);  
        }
        return res;
    }
};